xdsl/dialects/printf.py

```diff
@@ -85,9 +87,29 @@ def parse(cls: type[PrintFormatOp], parser: Parser) -> PrintFormatOp:
         return op
 
 
+@irdl_op_definition
+class PrintCharOp(IRDLOperation):
+    """
+    Print a single character
+
+    Equivalent to putchar in C.
+    """
+
+    name = "printf.print_char"
+    char: Operand = operand_def()
+
+
+@irdl_op_definition
+class PrintIntOp(IRDLOperation):
+    """
+    Print a single Integer
+    """
+
+    name = "printf.print_int"
+    int: Operand = operand_def(builtin.IntegerType)
```

Comment:
You added the `PrintIntOp` and had a justification in your PR description for it, but then you don't use it in the lowering afaics?

I would just drop it, as `print_int` is just a subset of `print_char`, right?

Reply:
It is used for sure, but i think i should write better filecheck tests for this

Reply:
Not sure why you consider `print_int` to be a subset of `print_char` I would actually consider it the other way around, `print_int` will actually generate multiple calls to `print_char`

Reply:
oooooh, print int prints more than just a single digit!